### PR TITLE
[controlles/job] Don't set minCount when PartialAdmission is disabled

### LINKED ISSUE #917: The job controller can potentially set `minCount` when  `features.PartialAdmission` is disabled
**What happened**:

If the job controller creates a workload specifying `minCont` for its podSet and `features.PartialAdmission` is not enabled, 
the mutation workload webhook will reset the `minCount` and make the workload `!equivalent` to the job.

**What you expected to happen**:

The job controller should not set the minCount if the featuregate is not enabled.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Don't set podSet's  minCount in job controller when the PartialAdmission feature gate is disabled.
#### Which issue(s) this PR fixes:

Fixes #917

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobs/job/job_controller.go [hunk 1]
```diff
@@ -311,9 +312,11 @@ func (j *Job) podsCount() int32 {
 }
 
 func (j *Job) minPodsCount() *int32 {
-	if strVal, found := j.GetAnnotations()[JobMinParallelismAnnotation]; found {
-		if iVal, err := strconv.Atoi(strVal); err == nil {
-			return pointer.Int32(int32(iVal))
+	if features.Enabled(features.PartialAdmission) {
```
Comment (alculquicondor) [inline pkg/controller/jobs/job/job_controller.go:315]: this should be checked in the framework code, not the job implementation.

**Comment by tenzen-y** [comment]
@trasc Can you update the release note? I think this PR is user-facing.

**Comment by alculquicondor** [comment]
I think we don't need a release note, as this feature was not released yet

**Comment by tenzen-y** [comment]
> I think we don't need a release note, as this feature was not released yet

makes sense.

File: pkg/controller/jobs/job/job_controller.go (hunk 1 above)
Comment (trasc) [reply pkg/controller/jobs/job/job_controller.go:315]: to do it in the framework we'll need to get the podsets and the set all mins to nil if the featuregate is not enabled ... in a couple of places

Comment (alculquicondor) [reply pkg/controller/jobs/job/job_controller.go:315]: I think that's the correct way.

Comment (trasc) [reply pkg/controller/jobs/job/job_controller.go]: ...  done

File: pkg/controller/jobframework/reconciler.go [hunk 2]
```diff
@@ -601,3 +602,14 @@ func (r *genericReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	return builder.Complete(r)
 }
+
+// resets the minCount for all podSets if the PartialAdmission feature is not enabled
+func restMinCounts(in []kueue.PodSet) []kueue.PodSet {
```
Comment (alculquicondor) [inline pkg/controller/jobframework/reconciler.go:607]:
```suggestion
func resetMinCountsIfDisabled(in []kueue.PodSet) []kueue.PodSet {
```

**Review by alculquicondor — COMMENTED** [review]
/approve

File: pkg/controller/jobframework/reconciler.go (hunk 2 above)
Comment (trasc) [reply pkg/controller/jobframework/reconciler.go:607]: the full version wold be too long , I'd need to specify what is disabled as well

Comment (alculquicondor) [reply pkg/controller/jobframework/reconciler.go:607]: then just `resetMinCounts`

Comment (trasc) [reply pkg/controller/jobframework/reconciler.go]: done

**Pull request merged**